### PR TITLE
Convert HABTM statuses/accounts tag associations to polymorphic HMT via `Taggable` concern

### PR DESCRIPTION
**Pull request by mjankowski**
Opening as draft for feedback, and because...

- If we can get this - https://github.com/mastodon/mastodon/pull/30654 - or something similar in first, some of these changes will be simpler/consolidated or done for us by the associations
- There are some other outstanding PRs which would clean up the queries within `Status`, and also simplify this PR in that respect.
- I believe the migration here is logically/datawise correct, but I've done zero contemplation about performance/duration/etc and it might need consideration on that front

### CODE REVIEW

File: db/migrate/20240731194619_create_taggings.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      t.bigint :account_id, null: false
+      t.bigint :tag_id, null: false
+      t.index [:account_id, :tag_id]
+    end
+  end
+
+  def convert_status_taggings
+    execute <<~SQL.squish
+      INSERT INTO taggings(tag_id, taggable_type, taggable_id, created_at, updated_at)
+      SELECT statuses_tags.tag_id, 'Status', statuses_tags.status_id, statuses.updated_at, statuses.updated_at
```
Comment (ziraqyoung) [inline db/migrate/20240731194619_create_taggings.rb:53]:
```suggestion
      SELECT statuses_tags.tag_id, 'Status', statuses_tags.status_id, statuses.created_at, statuses.updated_at
```

Comment (ziraqyoung) [reply db/migrate/20240731194619_create_taggings.rb:53]: Did you mean to update the `created_at` value to be the same as `updated_at`?

File: db/migrate/20240731194619_create_taggings.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      FROM statuses_tags
+      JOIN statuses ON statuses.id = statuses_tags.status_id
+    SQL
+  end
+
+  def convert_account_taggings
+    execute <<~SQL.squish
+      INSERT INTO taggings(tag_id, taggable_type, taggable_id, created_at, updated_at)
+      SELECT accounts_tags.tag_id, 'Account', accounts_tags.account_id, accounts.updated_at, accounts.updated_at
```
Comment (ziraqyoung) [inline db/migrate/20240731194619_create_taggings.rb:62]:
```suggestion
      SELECT accounts_tags.tag_id, 'Account', accounts_tags.account_id, accounts.created_at, accounts.updated_at
```

File: db/migrate/20240731194619_create_taggings.rb (hunk 1 above)
Comment (mjankowski) [reply db/migrate/20240731194619_create_taggings.rb:53]: > Did you mean to update the `created_at` value to be the same as `updated_at`?

I'm not sure what I meant ... and I also don't have a really strong opinion about which is more correct.

We currently dont have either datetime value on the habtm join table, but we will have it and be populating it going forward on the HMT taggings table. I think that backfilling the values with `NOW()` would be wrong, and I think either of the created_at or updated_at time for the account/status in question would be acceptable. Will gladly defer to anyone who feels strongly about one over the other.

Comment (ziraqyoung) [reply db/migrate/20240731194619_create_taggings.rb:53]: Oh I see, I was reading through this PR and I thought it was a typo. Thanks for the explanation.

**Pull request closed without merging**